diff -wpruN --no-dereference '--exclude=*.orig' a~/src/OpenSSL/crypto.py a/src/OpenSSL/crypto.py
--- a~/src/OpenSSL/crypto.py	1970-01-01 00:00:00
+++ a/src/OpenSSL/crypto.py	1970-01-01 00:00:00
@@ -947,7 +947,6 @@ utils.deprecated(
         "APIs in cryptography."
     ),
     DeprecationWarning,
-    name="X509Extension",
 )
 
 
@@ -2365,7 +2364,6 @@ utils.deprecated(
         "in cryptography."
     ),
     DeprecationWarning,
-    name="Revoked",
 )
 
 
@@ -2610,7 +2608,6 @@ utils.deprecated(
         "in cryptography."
     ),
     DeprecationWarning,
-    name="CRL",
 )
 
 
@@ -2863,7 +2860,6 @@ utils.deprecated(
     __name__,
     "sign() is deprecated. Use the equivalent APIs in cryptography.",
     DeprecationWarning,
-    name="sign",
 )
 
 
@@ -2910,7 +2906,6 @@ utils.deprecated(
     __name__,
     "verify() is deprecated. Use the equivalent APIs in cryptography.",
     DeprecationWarning,
-    name="verify",
 )
 
 
@@ -2952,7 +2947,6 @@ utils.deprecated(
         "in cryptography."
     ),
     DeprecationWarning,
-    name="dump_crl",
 )
 
 
@@ -2995,5 +2989,4 @@ utils.deprecated(
         "in cryptography."
     ),
     DeprecationWarning,
-    name="load_crl",
 )
