diff -wpruN --no-dereference '--exclude=*.orig' a~/src/OpenSSL/crypto.py a/src/OpenSSL/crypto.py
--- a~/src/OpenSSL/crypto.py	1970-01-01 00:00:00
+++ a/src/OpenSSL/crypto.py	1970-01-01 00:00:00
@@ -968,7 +968,6 @@ utils.deprecated(
         "APIs in cryptography."
     ),
     DeprecationWarning,
-    name="X509Extension",
 )
 
 
@@ -1203,7 +1202,6 @@ utils.deprecated(
         "in cryptography."
     ),
     DeprecationWarning,
-    name="X509Req",
 )
 
 
@@ -2400,7 +2398,6 @@ utils.deprecated(
         "in cryptography."
     ),
     DeprecationWarning,
-    name="Revoked",
 )
 
 
@@ -2645,7 +2642,6 @@ utils.deprecated(
         "in cryptography."
     ),
     DeprecationWarning,
-    name="CRL",
 )
 
 
@@ -2840,7 +2836,6 @@ utils.deprecated(
         "in cryptography."
     ),
     DeprecationWarning,
-    name="dump_certificate_request",
 )
 
 
@@ -2882,7 +2877,6 @@ utils.deprecated(
         "in cryptography."
     ),
     DeprecationWarning,
-    name="load_certificate_request",
 )
 
 
@@ -2926,7 +2920,6 @@ utils.deprecated(
     __name__,
     "sign() is deprecated. Use the equivalent APIs in cryptography.",
     DeprecationWarning,
-    name="sign",
 )
 
 
@@ -2973,7 +2966,6 @@ utils.deprecated(
     __name__,
     "verify() is deprecated. Use the equivalent APIs in cryptography.",
     DeprecationWarning,
-    name="verify",
 )
 
 
@@ -3015,7 +3007,6 @@ utils.deprecated(
         "in cryptography."
     ),
     DeprecationWarning,
-    name="dump_crl",
 )
 
 
@@ -3058,5 +3049,4 @@ utils.deprecated(
         "in cryptography."
     ),
     DeprecationWarning,
-    name="load_crl",
 )
