diff -wpruN --no-dereference '--exclude=*.orig' a~/Makefile.in a/Makefile.in
--- a~/Makefile.in	1970-01-01 00:00:00
+++ a/Makefile.in	1970-01-01 00:00:00
@@ -74,7 +74,7 @@ CFLAGS.icu = @CFLAGS_ICU@
 # When cross-compiling, we need to avoid the -s flag because it only
 # works on the build host's platform.
 INSTALL.strip.1 = $(INSTALL)
-INSTALL.strip.0 = $(INSTALL) -s
+INSTALL.strip.0 = $(INSTALL)
 INSTALL.strip   = $(INSTALL.strip.@IS_CROSS_COMPILING@)
 INSTALL.noexec  = $(INSTALL) -m 0644
 
