diff -wpruN --no-dereference '--exclude=*.orig' a~/tun.c a/tun.c
--- a~/tun.c	1970-01-01 00:00:00
+++ a/tun.c	1970-01-01 00:00:00
@@ -54,7 +54,7 @@ static	int  tunattach(dev_info_t *, ddi_
 static	int  tundetach(dev_info_t *, ddi_detach_cmd_t);
 static	int  tuninfo(dev_info_t *, ddi_info_cmd_t, void *, void **);
 static	int  tunopen(queue_t *, dev_t *, int, int, cred_t *);
-static	int  tunclose(queue_t *);
+static	int  tunclose(queue_t *, int, cred_t *);
 static	int  tunwput(queue_t *wq, mblk_t *mb);
 static	int  tunwsrv(queue_t *wq);
 
@@ -296,7 +296,7 @@ static int tunopen(queue_t *rq, dev_t *d
   return rc;
 }
 
-static int tunclose(queue_t *rq)
+static int tunclose(queue_t *rq, int flag, cred_t *cred)
 {
   struct tunstr **prev, *tmp, *str=(struct tunstr *)rq->q_ptr;
   struct tunppa *ppa;
