diff -wpruN --no-dereference '--exclude=*.orig' a~/tests/df/no-mtab-status.sh a/tests/df/no-mtab-status.sh
--- a~/tests/df/no-mtab-status.sh	1970-01-01 00:00:00
+++ a/tests/df/no-mtab-status.sh	1970-01-01 00:00:00
@@ -24,9 +24,6 @@ require_gcc_shared_
 # Protect against inaccessible remote mounts etc.
 timeout 10 df || skip_ "df fails"
 
-grep '^#define HAVE_GETMNTENT 1' $CONFIG_HEADER > /dev/null \
-      || skip_ "getmntent is not used on this system"
-
 # Simulate "mtab" failure.
 # Replace gnulib streq and C23 nullptr as that are not available here.
 sed 's/streq/0==str''cmp/; s/nullptr/NU''LL/' > k.c <<EOF || framework_failure_
@@ -35,7 +32,7 @@ sed 's/streq/0==str''cmp/; s/nullptr/NU'
 #include <stdlib.h>
 #include <errno.h>
 #include <fcntl.h>
-#include <mntent.h>
+#include <sys/mnttab.h>
 #include <string.h>
 #include <stdarg.h>
 #include <dlfcn.h>
