diff -wpruN --no-dereference '--exclude=*.orig' a~/gettext-tools/gnulib-m4/getaddrinfo.m4 a/gettext-tools/gnulib-m4/getaddrinfo.m4
--- a~/gettext-tools/gnulib-m4/getaddrinfo.m4	1970-01-01 00:00:00
+++ a/gettext-tools/gnulib-m4/getaddrinfo.m4	1970-01-01 00:00:00
@@ -108,7 +108,7 @@ int getaddrinfo (const char *, const cha
 #include <string.h>
             ]], [[
               struct addrinfo hints;
-              struct addrinfo ai;
+              struct addrinfo *ai;
               memset (&hints, 0, sizeof (hints));
               hints.ai_flags = AI_NUMERICSERV;
               return getaddrinfo ("www.gnu.org", "http", &hints, &ai) != EAI_NONAME;
