diff -wpruN --no-dereference '--exclude=*.orig' a~/gettext-runtime/intl/configure a/gettext-runtime/intl/configure
--- a~/gettext-runtime/intl/configure	1970-01-01 00:00:00
+++ a/gettext-runtime/intl/configure	1970-01-01 00:00:00
@@ -30647,6 +30647,7 @@ else case e in #(
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 #ifdef _MSC_VER
 #include <inttypes.h>
 /* See page about "Parameter Validation" on msdn.microsoft.com.
diff -wpruN --no-dereference '--exclude=*.orig' a~/libtextstyle/configure a/libtextstyle/configure
--- a~/libtextstyle/configure	1970-01-01 00:00:00
+++ a/libtextstyle/configure	1970-01-01 00:00:00
@@ -30457,6 +30457,7 @@ else case e in #(
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 #ifdef _MSC_VER
 #include <inttypes.h>
 /* See page about "Parameter Validation" on msdn.microsoft.com.
@@ -31406,6 +31407,7 @@ else case e in #(
 #include <signal.h>
 #include <stdio.h>
 #include <string.h>
+#include <unistd.h>
 #if HAVE_SNPRINTF
 # define my_snprintf snprintf
 #else
