--- a~/template.c	1970-01-01 00:00:00
+++ a/template.c	1970-01-01 00:00:00
@@ -203,7 +203,7 @@
         if (iLenVal > 0) { /* may be zero depending on property */
             /* first, make sure buffer fits */
             if (iBuf + iLenVal + extra_space >= iparam->lenBuf) /* we reserve one char for the final \0! */
-                CHKiRet(ExtendBuf(iparam, iBuf + iLenVal + 1));
+                CHKiRet(ExtendBuf(iparam, iBuf + iLenVal + extra_space));
 
             if (need_comma) {
                 memcpy(iparam->param + iBuf, ", ", 2);
