diff -wpruN --no-dereference '--exclude=*.orig' a~/termcap.c a/termcap.c
--- a~/termcap.c	1970-01-01 00:00:00
+++ a/termcap.c	1970-01-01 00:00:00
@@ -32,6 +32,7 @@
 
 #include <sys/types.h>
 #include <stdint.h>
+#include <strings.h>
 
 #include "screen.h"
 
diff -wpruN --no-dereference '--exclude=*.orig' a~/utmp.c a/utmp.c
--- a~/utmp.c	1970-01-01 00:00:00
+++ a/utmp.c	1970-01-01 00:00:00
@@ -39,6 +39,8 @@
 #include "screen.h"
 
 #ifdef HAVE_UTEMPTER
+#include <sys/types.h>
+#include <signal.h>
 #include <utempter.h>
 #endif
 
