--- a~/lib/nicInfo/nicInfoPosix.c.orig	Tue Aug  6 20:09:09 2024
+++ a/lib/nicInfo/nicInfoPosix.c	Tue Aug  6 20:09:16 2024
@@ -35,11 +35,8 @@
 #include <sys/stat.h>
 #include <errno.h>
 #include <limits.h>
-#if defined(__FreeBSD__) || defined(__APPLE__)
-# include <sys/sysctl.h>
 # include <ifaddrs.h>
 # include <net/if.h>
-#endif
 #ifndef NO_DNET
 # ifdef DNET_IS_DUMBNET
 #  include <dumbnet.h>
@@ -195,7 +192,7 @@
 }
 
 
-#if defined(USERWORLD) || defined(USE_SLASH_PROC) || (defined(__linux__) && defined(NO_DNET))
+#if defined(sun) || defined(USERWORLD) || defined(USE_SLASH_PROC) || (defined(__linux__) && defined(NO_DNET))
 /*
  ******************************************************************************
  * CountNetmaskBits --                                                   */ /**
@@ -234,7 +231,7 @@
 }
 #endif
 
-#if defined(USERWORLD) || (defined(__linux__) && defined(NO_DNET))
+#if defined(USERWORLD) || defined(sun) || (defined(__linux__) && defined(NO_DNET))
 static unsigned
 CountNetmaskBitsV6(struct sockaddr *netmask)
 {
@@ -306,7 +303,11 @@
     * records are intermingled with AF_INET and AF_INET6 records.
     */
    for (pkt = ifaddrs; pkt != NULL; pkt = pkt->ifa_next) {
+#ifdef sun
+      struct sockaddr_dl *sdl;
+#else
       struct sockaddr_ll *sll = (struct sockaddr_ll *)pkt->ifa_addr;
+#endif
 
       if (GuestInfo_IfaceGetPriority(pkt->ifa_name) != priority ||
           GuestInfo_IfaceIsExcluded(pkt->ifa_name)) {
@@ -313,7 +314,15 @@
          continue;
       }
 
+#ifdef sun
+      if (pkt->ifa_addr->sa_family == AF_LINK) {
+         uint8_t *mac;
+
+         sdl = (struct sockaddr_dl *)pkt->ifa_addr;
+         mac = LLADDR(sdl);
+#else
       if (sll != NULL && sll->sll_family == AF_PACKET) {
+#endif
          char macAddress[NICINFO_MAC_LEN];
          GuestNicV3 *nic;
          struct ifaddrs *ip;
@@ -324,7 +333,7 @@
           * and its ifa_flags is reported as 0. No AF_PACKET family ifaddrs
           * is reported for loopback interface.
           */
-#if !defined(USERWORLD)
+#if !defined(USERWORLD) && !defined(sun)
          /*
           * Ignore loopback and downed devices.
           */
@@ -335,8 +344,12 @@
 
          Str_Sprintf(macAddress, sizeof macAddress,
                      "%02x:%02x:%02x:%02x:%02x:%02x",
+#ifdef sun
+                     mac[0], mac[1], mac[2], mac[3], mac[4], mac[5]);
+#else
                      sll->sll_addr[0], sll->sll_addr[1], sll->sll_addr[2],
                      sll->sll_addr[3], sll->sll_addr[4], sll->sll_addr[5]);
+#endif
          nic = GuestInfoAddNicEntry(nicInfo, macAddress, NULL, NULL,
                                     maxNicsError);
          if (nic == NULL) {
@@ -452,7 +465,7 @@
    }
 
    return TRUE;
-#elif defined(USERWORLD) || defined(__linux__)
+#elif defined(USERWORLD) || defined(__linux__) || defined(sun)
    struct ifaddrs *ifaddrs = NULL;
 
    if (getifaddrs(&ifaddrs) == 0 && ifaddrs != NULL) {
@@ -501,6 +514,7 @@
  */
 #if defined(__FreeBSD__) || \
     defined(__APPLE__) || \
+    defined(sun) || \
     defined(USERWORLD) || \
     (defined(__linux__) && defined(NO_DNET))
 
