diff -wpruN --no-dereference '--exclude=*.orig' a~/meson.build a/meson.build
--- a~/meson.build	1970-01-01 00:00:00
+++ a/meson.build	1970-01-01 00:00:00
@@ -432,7 +432,6 @@ headers = [
   'sys/event.h',
   'sys/uio.h',
   'sys/filio.h',
-  'sys/inotify.h',
   'sys/mkdev.h',
   'sys/mntctl.h',
   'sys/mnttab.h',
@@ -731,7 +730,6 @@ functions = [
   'getvfsstat',
   'gmtime_r',
   'hasmntopt',
-  'inotify_init1',
   'issetugid',
   'kevent',
   'kqueue',
