diff -wpruN --no-dereference '--exclude=*.orig' a~/src/OpenSSL/crypto.py a/src/OpenSSL/crypto.py
--- a~/src/OpenSSL/crypto.py	1970-01-01 00:00:00
+++ a/src/OpenSSL/crypto.py	1970-01-01 00:00:00
@@ -2840,8 +2840,7 @@ utils.deprecated(
         "PKCS#12 support in pyOpenSSL is deprecated. You should use the APIs "
         "in cryptography."
     ),
-    DeprecationWarning,
-    name="PKCS12",
+    DeprecationWarning
 )
 
 
