diff -wpruN --no-dereference '--exclude=*.orig' a~/libexslt/date.c a/libexslt/date.c
--- a~/libexslt/date.c	1970-01-01 00:00:00
+++ a/libexslt/date.c	1970-01-01 00:00:00
@@ -46,6 +46,7 @@
 #include "exslt.h"
 
 #include <string.h>
+#include <stdlib.h>
 
 #ifdef HAVE_ERRNO_H
 #include <errno.h>
diff -wpruN --no-dereference '--exclude=*.orig' a~/libxslt/extensions.c a/libxslt/extensions.c
--- a~/libxslt/extensions.c	1970-01-01 00:00:00
+++ a/libxslt/extensions.c	1970-01-01 00:00:00
@@ -14,6 +14,7 @@
 
 #include <string.h>
 #include <limits.h>
+#include <stdlib.h>
 
 #include <libxml/xmlmemory.h>
 #include <libxml/tree.h>
@@ -2247,7 +2248,7 @@ xsltRegisterTestModule(void)
 static void
 xsltHashScannerModuleFree(void *payload ATTRIBUTE_UNUSED,
                           void *data ATTRIBUTE_UNUSED,
-                          xmlChar * name ATTRIBUTE_UNUSED)
+                          const xmlChar * name ATTRIBUTE_UNUSED)
 {
 #ifdef WITH_MODULES
     xmlModuleClose(payload);
